adding user#create

mmmaaatttttt opened the pull request:
hey dude, here's my attempt to add signup. I want signup to also effectively log in, so i moved the authenticate method to the application controller so that it can be shared. can you think of a better way?

mmmaaatttttt committed:
adding user#create

tigarcia commented on test/controllers/user_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
delete me

tigarcia commented on app/helpers/user_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module UserHelper
```
delete

tigarcia commented on app/controllers/application_controller.rb:
```diff
@@ -1,10 +1,20 @@
 
   before_action :authenticate_request
 
+  def authenticate
```
Maybe you could make the method take 2 parameters for email and password instead of using the http params.  I was confused about how it was working at first.  You'd have to change the route for login

mmmaaatttttt committed:
removing unnecessary files, refactoring authenticate

authenticate is now a function of two parameters;
authentication_controller now uses super to call parent method.

mmmaaatttttt committed:
fix bug on favorites page

links on the favorites page are json links to the story,
in the rails app not the original link from the HN story.
Modifying the json builder fixes this.

mmmaaatttttt commented:
updated with your suggestions. also for some reason in my version the links were all busted on my favorites.html page, i modified the json builder to fix this.
